components/search/src/selector.rs

```diff
@@ -2,20 +2,31 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use crate::{error::Error, RefinedSearchConfig, SearchApiResult, SearchUserEnvironment};
+use crate::filter::filter_engine_configuration;
+use crate::{
+    error::Error, JSONSearchConfiguration, RefinedSearchConfig, SearchApiResult,
+    SearchUserEnvironment,
+};
 use error_support::handle_error;
+use parking_lot::Mutex;
+use std::sync::Arc;
+
+#[derive(Default)]
+pub(crate) struct SearchEngineSelectorInner {
+    configuration: Option<JSONSearchConfiguration>,
+}
 
 /// SearchEngineSelector parses the JSON configuration for
 /// search engines and returns the applicable engines depending
 /// on their region + locale.
 #[derive(Default, uniffi::Object)]
-pub struct SearchEngineSelector {}
+pub struct SearchEngineSelector(Mutex<SearchEngineSelectorInner>);
 
 #[uniffi::export]
 impl SearchEngineSelector {
     #[uniffi::constructor]
     pub fn new() -> Self {
-        Self::default()
+        Self(Mutex::new(SearchEngineSelectorInner::default()))
```

Comment:
```suggestion
        Self(Mutex::default())
```

(`Mutex<T>` implements `Default` if `T` implements `Default`—but your code looks good as-is, too, if you'd like to keep it explicit!)